lib/src/conflicts.rs

```diff
@@ -196,17 +196,18 @@ pub fn materialize_merge_result(
                         output.write_all(CONFLICT_START_LINE)?;
                         let mut add_index = 0;
                         for left in &removes {
-                            if add_index == adds.len() {
+                            let right1 = if let Some(right1) = adds.get(add_index) {
```

Comment:
When our MSRV is high enough (1.65) we can rewrite this with a `let else` since the `else` part diverges.

Reply:
Yes, and some other nested `if let` can be replaced with early return.